### PR TITLE
Allow custom User Agent strings (close #23)

### LINKED ISSUE #23: Allow custom User Agent strings
**Is your feature request related to a problem? Please describe.**
Currently the UA Strings are hardcoded on lines:

https://github.com/snowplow/snowplow-cpp-tracker/blob/master/src/http_client.cpp#L72
https://github.com/snowplow/snowplow-cpp-tracker/blob/master/src/http_client.cpp#L193

We should introduce the ability to allow the user to override these.

This is especially important as the YAUAA enrichment currently identifies any events from `Win32` as Windows 95. Better User Agent strings (such as including Windows NT 10.0) would allow for more accurate device identification.

### PR DESCRIPTION
**Pull request by matus-tomlein**
Adds a `set_useragent()` function to Subject to set a custom useragent by configuring the `ua` property in events.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: src/subject.hpp [hunk 1]
```diff
@@ -32,6 +32,7 @@ class Subject {
   void set_color_depth(int depth);
   void set_timezone(const string & timezone);
   void set_language(const string & language);
+  void set_useragent(const string & user_agent);
```
Comment (AlexBenny) [inline src/subject.hpp:35]: Funny to see that different languages have different styles to do the same thing.
Pointer operators in ObjC are suggested near the var name (e.g.: `NSString *var`) meanwhile in C++ seems preferred near the type (e.g.: `string* p`). Here we placed it in the middle. Maybe it was the trend back in the 2016 :)

**Review by AlexBenny — COMMENTED** [review]
Looks good!
My doubt is if we should apply that to the HttpClient too: https://github.com/snowplow/snowplow-cpp-tracker/blob/08e762abad82c1f499dc719549016c18b3776dad/src/http_client.cpp#L193

I guess if we set it in the event we could probably use the same in the http request. wdyt?

**Comment by matus-tomlein** [comment]
@AlexBenny You're right, but it would require a bit of a bigger change in the code to propagate the Subject configuration for useragent and read it in the HttpClient when creating the HTTP request. Currently neither the Emitter nor the HttpClient are aware of the subject properties (they are only given the payload). I think it's not worth the effort to change the header because it doesn't have an effect on the useragent anyway – the one in the "ua" property takes precedence.

**Review by AlexBenny — APPROVED** [review]

**Pull request merged**